### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by shihabrafio**
- Remove `current_user` method in `ApplicationController`, devise will provide us with one.
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- Modify the views in devisefor registration and login to match the wireframes from the Sneak Peek and your styling.

### CODE REVIEW

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @shihabrafio 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🔴 🔴 🔴 🔴 🔴 🔴 

You have submitted a project, that does not have a contribution from your coding partner. please contact them and collaborate with them as required by this milestone. after that you can ask for a review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class User < ApplicationRecord
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
```
Comment (Qoosim) [inline app/models/user.rb:3]: - [ ] Great job team for adding devise to your app. However, some things are not working as supposed.
First, while setting up your app on my local machine with this command `rails db:setup` which would create the db on my pc. It fires up confirm your email page on the browser which is not normal. That page is supposed to display after clicking on the `sign up` button on the sign up page to confirm the user's email. Kindly look into your setup properly and fix where the issue is.

### 
your-setup

###
your-confirm-page

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" ,class: "bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 mt-4 rounded mb-10"%>
+  </div>
+<% end %>
+<%= render "devise/shared/links" %>
```
Comment (Qoosim) [inline app/views/devise/sessions/new.html.erb:26]: - [x] Name input field is missing on the page and there's error message popped while trying to register a user. Take a look.

###
your-signup

###
your-signup-page-2

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Implemented some requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by divinecharlotte — APPROVED** [review]
# Hello @Team      😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 
### To Highlight 💯

- Good implementation of the required changes ✔️
- Descriptive Readme ✔️






## Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**